### ISSUE TITLE
[IATR](M0) Fetch run data with failed tests from Cloud

### ISSUE THREAD

**Issue by warrensplayer** [cypress-io/cypress#24440]
Use the new GraphQL queries from the Cloud to fetch the data to populate the new "Failures" page

### Description
The failures page will show test failures from the "most relevant" run to the current local environment.  This will be determined by sending the last 100 git commit hashes to request related `CloudRun` information with the cloud GraphQL field `relatedRunsForGitCommits` developed [here](https://cypress-io.atlassian.net/browse/CYCLOUD-666) and then requesting the failed specs for the most recent run with the GraphQL field `testFailuresByRunId` developed [here](https://cypress-io.atlassian.net/browse/CYCLOUD-667).

### Out of scope
* For M0, only the new Failures page will make use of these new fields. The cloud team is performing a spike [here](https://cypress-io.atlassian.net/browse/CYCLOUD-668) to look at what it will take to use the same logic to replace the algorithm currently used to determine the values for the `RunStatusDots` on the SpecsList page. The implementation of those changes will be in a future milestone for IATR.
* A future milestone will introduce a badge shown in the left hand navigation within the "Failures" nav item to indicate if failures are present.

### Technical Requirements

* Leverage the pattern that was recently established to consolidate cloud status determination to the shared root component `CloudViewerAndProject`.
* Query for and save the relevant run information in the store used in the `CloudViewAndProject` component for use in the Failures page for this milestone. (A future milestone will expand the use case to store the 4 most relevant runs. See "Out of scope" above.)
* When visiting the "Failures" page, the relevant run information will be fetched from the store and used to query for the failed specs information to display on the page.

Cloud Epic: https://cypress-io.atlassian.net/browse/CYCLOUD-665

**Comment by warrensplayer** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Component-Testing-606e43bbf1b3b20013c142c1/issues/cypress-io/cypress/24440?planning-poker) with Zenhub @rockindahizzy @lmiller1990 @ZachJW34 @marktnoonan @amehta265 @astone123

**Comment by lmiller1990** [comment]
I'm adding a bit of buffer to my vote. The way we currently test GraphQL calls is a bit messy, confusing and doesn't give me much confidence. I'd like to see some thought put into how we can effectively test and develop around the App <> Cloud seam. I also think testing for reviewers is going to be somewhat time consuming, so that's also included in my vote.

**Comment by warrensplayer** [comment]
Blocked while waiting on completion of the related Cloud issues.
